include/caffe/common.hpp

```diff
@@ -149,6 +149,10 @@ class Caffe {
   static void SetDevice(const int device_id);
   // Prints the current GPU status.
   static void DeviceQuery();
+  // Check if specified device is available
+  static bool CheckDevice(const int device_id);
+  // Get the first available device id since start_id
```

Comment:
"that's at least start_id", right? "Since" is ambiguous as to inclusion (and doesn't read well anyway).